### PR TITLE
[risk=low][RW-7748] Use mariadb 10.2 instead of mysql docker images

### PR DESCRIPTION
**Pull request by jmthibault79**
A suggested solution for the Apple-M1 docker mysql problem: https://docs.docker.com/desktop/mac/apple-silicon/

but dev-up results in an invalid password format.  After attempting to supply a password in a valid format (converted "root-notasecret", see second commit) dev-up now results in
```
Database init & migrations...
+ ./run-migrations.sh main
+ IFS='
        '
+ activity=-PrunList=main
+ '[' -z ']'
+ context=
+ CREATE_DB_FILE=/tmp/create_db.sql
+ trap finish EXIT
+ envsubst
++ dirname ./run-migrations.sh
+ echo 'Creating database if it does not exist...'
Creating database if it does not exist...
+ cat /tmp/create_db.sql
+ run_mysql -h 127.0.0.1 --port 3306 -u root -p388A9C183778FC61387343567C2C6CFCEF6F0876
+ '[' -f /.dockerenv ']'
+ echo 'Outside docker: invoking mysql via docker for portability'
Outside docker: invoking mysql via docker for portability
+ docker run --rm --network host --entrypoint '' -i mariadb:10.2 mysql -h 127.0.0.1 --port 3306 -u root -p388A9C183778FC61387343567C2C6CFCEF6F0876
ERROR 2002 (HY000): Can't connect to MySQL server on '127.0.0.1' (115)
+ finish
+ rm -f /tmp/create_db.sql
+ docker-compose down
Stopping api_db_1 ... done
Removing api_db_1 ... done
Removing network api_default

```

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Comment by calbach** [comment]
@jmthibault79 I hope you don't mind, I pushed some more changes here. This branch now works locally for me - would need to be verified on M1 though.

File: api/db/create_db.sql [hunk 1]
```diff
@@ -4,9 +4,9 @@ CREATE USER IF NOT EXISTS '${DEV_READONLY_DB_USER}'@'%' IDENTIFIED BY '${DEV_REA
 CREATE USER IF NOT EXISTS '${WORKBENCH_DB_USER}'@'%' IDENTIFIED BY '${WORKBENCH_DB_PASSWORD}';
 CREATE USER IF NOT EXISTS '${LIQUIBASE_DB_USER}'@'%' IDENTIFIED BY '${LIQUIBASE_DB_PASSWORD}';
 -- In case the users already exist, change their passwords.
-SET PASSWORD FOR '${DEV_READONLY_DB_USER}'@'%' = '${DEV_READONLY_DB_PASSWORD}';
-SET PASSWORD FOR '${WORKBENCH_DB_USER}'@'%' = '${WORKBENCH_DB_PASSWORD}';
-SET PASSWORD FOR '${LIQUIBASE_DB_USER}'@'%' = '${LIQUIBASE_DB_PASSWORD}';
+SET PASSWORD FOR '${DEV_READONLY_DB_USER}'@'%' = PASSWORD('${DEV_READONLY_DB_PASSWORD}');
```
Comment (calbach) [inline api/db/create_db.sql:7]: This resolved the password hashing complaint.

File: api/libproject/devstart.rb [hunk 2]
```diff
@@ -80,9 +80,11 @@ def start_local_db_service()
   bm = Benchmark.measure {
     common.run_inline %W{docker-compose up -d db}
 
+    root_pass = Workbench.read_vars_file("db/vars.env")["MYSQL_ROOT_PASSWORD"]
+
     common.status "waiting up to #{deadlineSec}s for mysql service to start..."
     start = Time.now
-    until (common.run %W{docker-compose exec -T db mysqladmin ping --silent}).success?
+    until (common.run "docker-compose exec -T db mysql -p#{root_pass} --silent -e 'SELECT 1;'").success?
```
Comment (calbach) [inline api/libproject/devstart.rb:87]: For some reason `mysqladmin ping` seemingly was returning success before the database was actually ready. This resulted in:

```
ERROR 2013 (HY000): Lost connection to MySQL server at 'handshake: reading initial communication packet', system error: 11
```

on the first run after clearing your DB. Issuing a no-op SQL command SELECT 1 seems to work more reliably

**Comment by jmthibault79** [comment]
Works great for me on M1.  Thanks!

**Comment by jmthibault79** [comment]
Also works for my Intel-Mac and it seems to have done an in-place upgrade.

Switching back to a mysql branch was not compatible with this new DB - but I didn't really expect it to be

**Review by calbach — APPROVED** [review]

**Pull request merged**